aks.tf

```diff
@@ -34,21 +43,19 @@ resource "azurerm_kubernetes_cluster" "aks" {
   api_server_authorized_ip_ranges = var.api_server_authorized_ip_ranges
 
   default_node_pool {
-    enable_node_public_ip = local.node_pools.platform.enable_node_public_ip
-    name                  = "platform"
-    node_count            = local.node_pools.platform.initial_count
-    node_labels           = local.node_pools.platform.node_labels
-    vm_size               = local.node_pools.platform.vm_size
-    zones                 = local.node_pools.platform.zones
-    os_disk_size_gb       = local.node_pools.platform.os_disk_size_gb
-    node_taints           = local.node_pools.platform.node_taints
-    enable_auto_scaling   = local.node_pools.platform.enable_auto_scaling
-    min_count             = local.node_pools.platform.min_count
-    max_count             = local.node_pools.platform.max_count
-    max_pods              = local.node_pools.platform.max_pods
-    tags                  = var.tags
+    enable_node_public_ip        = false
+    only_critical_addons_enabled = true
```

Comment:
[only_critical_addons_enabled](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#only_critical_addons_enabled) - (Optional) Enabling this option will taint default node pool with CriticalAddonsOnly=true:NoSchedule taint. Changing this forces a new resource to be created.

Reply:
Azure documentation:
```
Use the CriticalAddonsOnly=true:NoSchedule taint to prevent application pods from being scheduled on system node pools.
```